### PR TITLE
Remove `_call_accelerator_hook` Trainer method

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?

After #10890 we only have one remaining usage of `_call_accelerator_hook`. We can remove this call by moving the line `torch.cuda.empty_cache()` from `on_train_start` to `setup` in `accelerators/gpu.py`. Then we can remove the `_call_accelerator_hook` Trainer method entirely.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/accelerators/gpu.py [hunk 1]
```diff
@@ -45,12 +45,9 @@ def setup_environment(self) -> None:
 
     def setup(self, trainer: "pl.Trainer") -> None:
         self.set_nvidia_flags(trainer.local_rank)
-        return super().setup(trainer)
-
-    def on_train_start(self) -> None:
-        super().on_train_start()
         # clear cache before training
         torch.cuda.empty_cache()
+        return super().setup(trainer)
```
Comment (four4fish) [inline pytorch_lightning/accelerators/gpu.py:50]: the original order is call super first then empty_cache?
This will matter more soon as Adrian is working on moving accelerator into ttp, the setup function will merge then.

Comment (daniellepintz) [reply pytorch_lightning/accelerators/gpu.py]: good call, fixed. also removed the "return" as I don't think we need to return anything

**Comment by daniellepintz** [comment]
@four4fish why is this a breaking change? It doesn't affect users

**Comment by four4fish** [comment]
> @four4fish why is this a breaking change? It doesn't affect users

for customized accelerator user who may override this function

**Review by ananthsub — APPROVED** [review]
please be sure to fill out the checklist (especially adding a changelog entry) before this is merged

File: CHANGELOG.md [hunk 2]
```diff
@@ -239,6 +239,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `HorovodPlugin.start_{training,evaluating,predicting}` hooks ([#10989](https://github.com/PyTorchLightning/pytorch-lightning/pull/10989))
 
 
+- Removed `Trainer._call_accelerator_hook` ([#10999](https://github.com/PyTorchLightning/pytorch-lightning/pull/10999))
```
Comment (ananthsub) [inline CHANGELOG.md:242]: `_call_accelerator_hook` doesn't need to be mentioned. this was added in between minor versions and was never part of an official release. 

the changelog should mention that `Accelerator.on_train_start` was removed

Comment (daniellepintz) [reply CHANGELOG.md]: oh, makes sense. fixed

**Review by awaelchli — APPROVED** [review]
